### PR TITLE
Validation & flash to posts and users & model specs

### PR DESCRIPTION
**Pull request by tamoorsaeed22**
### Project requirements

- Add the following validations:

> For the User model:

- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

> For the Post model:

- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

- Add unit specs for all of your models' methods and validations.
- Add flash messages in the create actions in all your controllers.
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe
[
Screenshot (50)
](url)
[
Screenshot (51)
](url)

### CODE REVIEW

File: spec/models/user_spec.rb
Comment (fobadara) [inline spec/models/user_spec.rb:1]: [REQUIRED] It is a requirement to include tests for custom methods. Kindly include tests for the recent_posts method.

File: spec/models/post_spec.rb
Comment (fobadara) [inline spec/models/post_spec.rb:1]: [REQUIRED] It is a requirement to include tests for custom methods. Kindly include tests for recent_comments and update_author_counter.

File: spec/models/comment_spec.rb
Comment (fobadara) [inline spec/models/comment_spec.rb:1]: [REQUIRED] It is a requirement to include tests for custom methods. Kindly include test for the update_counter method

File: spec/models/like_spec.rb
Comment (fobadara) [inline spec/models/like_spec.rb:1]: [REQUIRED] It is a requirement to include tests for custom methods. Kindly include test for the update_counter component.

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @tamoorsaeed22, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

REDO

## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt
✅ Good implementation of the principle of separating concerns.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tamoorsaeed22** [comment]
@fobadara Thank you for correcting me I have added all the required custom methods tests and

-  update_counter method
-  update_counter component.
- recent_comments and update_author_counter.
- recent_posts method

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @tamoorsaeed22, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] All model specs pass with green checks :white_check_mark: 
- [x] Git flow is setup correctly :white_check_mark:  
- [x] Model validations have been set up correctly :white_check_mark:  
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:   
- [x] Implemented all the changes from the previous reviewer :white_check_mark:  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**